api/search.go

```diff
@@ -0,0 +1,39 @@
+package api
+
+import (
+	"github.com/hashicorp/nomad/api/contexts"
+)
+
+type Search struct {
+	client *Client
+}
+
+// Search returns a handle on the Search endpoints
+func (c *Client) Search() *Search {
+	return &Search{client: c}
+}
+
+// PrefixSearch returns a list of matches for a particular context and prefix. If a
+// context is not specified, matches for all contexts are returned.
```

Comment:
Nit: Might be worth deleting the last line of the comment since we now assume there will always be a context, just `contexts.All`